core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -418,7 +418,7 @@ class BlockManagerMasterEndpoint(
     if (externalShuffleServiceRemoveShuffleEnabled) {
       mapOutputTracker.shuffleStatuses.get(shuffleId).foreach { shuffleStatus =>
         shuffleStatus.withMapStatuses { mapStatuses =>
-          mapStatuses.foreach { mapStatus =>
+          mapStatuses.filter(_ != null).foreach { mapStatus =>
```

Comment:
It means that there's explicit `Some(null)` which we should avoid.

Reply:
@HyukjinKwon it's not an `Option` but an `Array`

```
val mapStatuses = new Array[MapStatus](numPartitions)
```
I think the fix is right.